core/domain/event_services.py

```diff
@@ -93,6 +94,57 @@ def _handle_event(
                 rule_spec_index, classification_categorization, params,
                 session_id, time_spent_in_secs))
 
+        is_feedback_useful = True
+        if classification_categorization == (
+                exp_domain.DEFAULT_OUTCOME_CLASSIFICATION):
+            is_feedback_useful = False
+
+        stats_models.AnswerSubmittedEventLogEntryModel.create(
+            exploration_id, exploration_version, state_name, session_id,
+            time_spent_in_secs, is_feedback_useful)
+
+        update_params = {
+            'is_feedback_useful': is_feedback_useful
+        }
+        if feconf.ENABLE_NEW_STATS_FRAMEWORK:
+            stats_services.update_stats(
+                exploration_id, exploration_version, state_name, cls.EVENT_TYPE,
+                update_params)
+
+
+class ExplorationActualStartEventHandler(BaseEventHandler):
+    """Event handler for recording exploration actual start events."""
+
+    EVENT_TYPE = feconf.EVENT_TYPE_ACTUAL_START_EXPLORATION
+
+    @classmethod
+    def _handle_event(
+            cls, exp_id, exp_version, state_name, session_id,
+            client_time_spent_in_secs):
+        stats_models.ExplorationActualStartEventLogEntryModel.create(
```

Comment:
ditto, why not under "new stats framework"? Shouldn't make any changes to data storage until we're ready to turn the whole thing on. Otherwise we have less freedom later to change this if we find out that, for some reason, we've designed it wrong.

Same comment applies elsewhere.

Reply:
Done.